lib/puppet/type/user.rb

```diff
@@ -1,6 +1,5 @@
 # frozen_string_literal: true
 
-require 'etc'
```

Comment:
One subtly is other user providers may be relying on this require without knowing it, for example `hpux`

https://github.com/puppetlabs/puppet/blob/ca922ca1f7c1cbd8c28b35a59c126b39b2c90778/lib/puppet/provider/user/hpux.rb#L51

Also the base `NameProvider` which is extended by user and group providers depends on our Etc wrapper: https://github.com/puppetlabs/puppet/blob/ca922ca1f7c1cbd8c28b35a59c126b39b2c90778/lib/puppet/provider/nameservice.rb#L30

And our wrapper calls `Etc` methods without requiring 'etc' https://github.com/puppetlabs/puppet/blob/ca922ca1f7c1cbd8c28b35a59c126b39b2c90778/lib/puppet/etc.rb

Maybe we should look at these call sites:

```
$ git grep '^[^#:]*Etc' lib
lib/puppet/etc.rb:      @password_class ||= Struct.new(*Etc::Passwd.members, *Etc::Passwd.members.map { |member| "canonical_#{member}".to_sym })
lib/puppet/etc.rb:      @group_class ||= Struct.new(*Etc::Group.members, *Etc::Group.members.map { |member| "canonical_#{member}".to_sym })
lib/puppet/etc.rb:      new_struct = struct.is_a?(Etc::Passwd) ? puppet_etc_passwd_class.new : puppet_etc_group_class.new
lib/puppet/feature/base.rb:  !Etc.getpwuid(0).nil? && Puppet.features.syslog?
lib/puppet/file_system/uniquefile.rb:  @@systmpdir ||= defined?(Etc.systmpdir) ? Etc.systmpdir : '/tmp'
lib/puppet/provider/file/posix.rb:      user = Etc.getpwuid(id)
lib/puppet/provider/file/posix.rb:      group = Etc.getgrgid(id)
lib/puppet/provider/group/groupadd.rb:        Etc.getpwnam(user.strip)
lib/puppet/provider/nameservice.rb:              "Cannot determine Etc section without a resource type"
lib/puppet/provider/user/hpux.rb:    while ent = Etc.getpwent() # rubocop:disable Lint/AssignmentInCondition
lib/puppet/provider/user/hpux.rb:    Etc.endpwent()
lib/puppet/provider/user/hpux.rb:    ent = Etc.getpwnam(resource.name)
lib/puppet/type/exec.rb:      Etc.getpwuid(Process.uid).name
lib/puppet/util/posix.rb:      Etc.send(method, id).send(field)
lib/puppet/util/posix.rb:    Etc.send(type) do |object|
lib/puppet/util/suidmanager.rb:      gid = Etc.getpwuid(uid).gid
lib/puppet/util/suidmanager.rb:    pwent = Etc.getpwuid(uid)
```

And make sure each of those files contain `require 'etc'`?